temporal-sdk/src/main/java/io/temporal/internal/common/GrpcUtils.java

```diff
@@ -14,4 +18,20 @@ public static boolean isChannelShutdownException(StatusRuntimeException ex) {
         && (description.startsWith("Channel shutdown")
             || description.startsWith("Subchannel shutdown")));
   }
+
+  public static <T> CompletableFuture<T> toCompletableFuture(ListenableFuture<T> listenableFuture) {
+    CompletableFuture<T> result = new CompletableFuture<>();
+    listenableFuture.addListener(
+        () -> {
+          try {
+            result.complete(listenableFuture.get());
+          } catch (ExecutionException e) {
+            result.completeExceptionally(e.getCause());
+          } catch (Exception e) {
+            result.completeExceptionally(e);
+          }
+        },
+        ForkJoinPool.commonPool());
```

Comment:
This is also how we convert grpc futures to completable futures for the client so I copied the same approach here.